File: QuanlyNhahang_API/Model/Item.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace QuanlyNhahang_API.Model
{
    public class Item
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Price { get; set; }
        public double Discount { get; set; }
        public int Quatity { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public bool Deleted { get; set; }
        public virtual Restaurant Restaurant { get; set; }
        public virtual User CreatedUser { get; set; }
        public virtual User UpdatedUser { get; set; }
        public virtual Category Category { get; set; }
        public virtual ItemImage ItemImage { get; set; }

    }
}

File: src/app/model/item.ts
import { Category } from "./category";
import { CategoryType } from "./categoryType";
import { Image } from "./image";
import { User } from "./user";

export interface Item{
    id:number,
    name: string,
    description:string,
    price:string,
    created:Date,
    updated:Date,
    deleted:boolean,
    discount:number;
    category:{
        id:number,
        name: string,
        description:string,
        created:Date,
        updated:Date,
        deleted:boolean,
        categoryType:CategoryType,
    },
    itemImage:{
        id:number;
        name: string;
        data:string,
        description:string;
        created:Date;
        updated:Date;
        deleted:boolean;
    }
    restaurant:{
        id:number;
        name: string;
        description:string;
        phone:string;
        address:string;
        created:Date;
        updated:Date;
        deleted:boolean;
    }
    createdUser:User["createdUser"],
   updatedUser:User["updatedUser"],
}